### PR TITLE
Second pixel testing update

### PR DESCRIPTION
**Pull request by mvollmer**
https://mvollmer.github.io/cockpit-project.github.io/blog/pixel-mocking.html

### CODE REVIEW

File: _posts/2023-09-01-pixel-mocking.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+title: Cockpit 299
+author: mvo
+date: '2023-09-01'
+tags: cockpit
+slug: pixel-mocking
+category: tutorial
+summary: 'Mocking the DOM for Better Pixels, plus statistics'
+---
+
+When testing pixels, once needs to deal with parts of the UI that
```
Comment (jelly) [inline _posts/2023-09-01-pixel-mocking.md:11]: `once` -> one?

File: _posts/2023-09-01-pixel-mocking.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+elements.
+
+For example, if a DOM element that should be ignored gets smaller from
+one run to the next, the old pixels in the reference image that are
+now outside of this element (but used to be inside) will now count as
+a difference.
+
+In reality, this is much less of a problem than it might seem: We
+usually compare big containers for pixel tests, such as a whole dialog
+or top-level panels. These larger elements are not supposed to chanmge
```
Comment (jelly) [inline _posts/2023-09-01-pixel-mocking.md:45]: cahnmge => change

File: _posts/2023-09-01-pixel-mocking.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+size when thei content changes.
```
Comment (jelly) [inline _posts/2023-09-01-pixel-mocking.md:46]: thei => their

File: _posts/2023-09-01-pixel-mocking.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+So we started exploring another approach: Replacing the real data in
+the UI with "mock data". This is conceptually simple, but I got a
+serious case of "perfection is the enemy of the good". I was imagining
+all kinds of theoretical problems, and I wanted to avoid them all with
+a bullet proof implementation. This failed, but the journey might be
+worth telling.
+
+The main issue is that the DOM is "owned" by React, and modifiying it
+without involcing React seemed like asking for trouble. React might
```
Comment (jelly) [inline _posts/2023-09-01-pixel-mocking.md:64]: involving

**Review by jelly — CHANGES_REQUESTED** [review]

File: _posts/2023-09-01-pixel-mocking.md (hunk 1 above)
Comment (mvollmer) [reply _posts/2023-09-01-pixel-mocking.md]: Thanks! I should have made this draft, I didn't actually proof read this myself in the beginning...

**Comment by martinpitt** [comment]
@mvollmer This got stuck -- do you mind checking if this is still up to date? Then let's bump the time stamp and publish it.

**Pull request closed without merging**